CodeGeneration/Sources/generate-swiftsyntax/templates/swiftsyntax/SwiftSyntaxCompatibilityFile.swift

```diff
@@ -0,0 +1,135 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftSyntax
+import SwiftSyntaxBuilder
+import SyntaxSupport
+import Utils
+
+let swiftSyntaxCompatibilityFile = try! SourceFileSyntax(leadingTrivia: copyrightHeader) {
+  for layoutNode in SYNTAX_NODES.compactMap(\.layoutNode).filter({ $0.children.hasDeprecatedChild }) {
+    try ExtensionDeclSyntax("extension \(raw: layoutNode.type.syntaxBaseName)") {
+      for child in layoutNode.children {
+        if let deprecatedName = child.deprecatedName {
+          let childType: TypeSyntax = child.kind.isNodeChoicesEmpty ? child.syntaxNodeKind.syntaxType : "\(raw: child.name)"
+          let type = child.isOptional ? TypeSyntax("\(raw: childType)?") : TypeSyntax("\(raw: childType)")
+
+          DeclSyntax(
+            """
+            @available(*, deprecated, renamed: "\(raw: child.varName)")
+            public var \(raw: deprecatedName.withFirstCharacterLowercased): \(raw: type) {
```

Comment:
Can’t you use `deprecatedVarName` here? Similar below. I would like to keep all the `withFirstCharacterLowercased` etc. calls in Child.swift and just use somewhat semantic accessors like `deprecatedVarName` here.